src/tests/ftest/launch.py

```diff
@@ -52,2294 +68,2973 @@
+            depth (int): max depth for find command
+            timeout (int): number of seconds to wait for the command to complete.
 
-    Args:
-        hosts (NodeSet): list of hosts on which to stop the service.
-        service (str): name of the service
+        Returns:
+            int: status code: 0 = success, 16 = failure
 
-    Returns:
-        int: status code: 0 = success, 512 = failure
+        """
+        logger.debug("-" * 80)
+        logger.debug("Moving files from %s to %s on %s", source, destination, hosts)
+
+        # Core and dump files require a file ownership change before they can be copied
+        if "stacktrace" in destination or "daos_dumps" in destination:
+            # pylint: disable=import-outside-toplevel
+            other = ["-print0", "|", "xargs", "-0", "-r0", "sudo", "-n", get_chown_command()]
+            if not run_remote(logger, hosts, find_command(source, pattern, depth, other)).passed:
+                message = f"Error changing {os.path.join(source, pattern)} file permissions"
+                self._fail_test(self.result.tests[-1], "Process", message)
+                return 16
+
+        # Use the last directory in the destination path to create a temporary sub-directory on the
+        # remote hosts in which all the source files matching the pattern will be copied. The entire
+        # temporary sub-directory will then be copied back to this host and renamed as the original
+        # destination directory plus the name of the host from which the files originated. Finally
+        # delete this temporary sub-directory to remove the files from the remote hosts.
+        rcopy_dest, tmp_copy_dir = os.path.split(destination)
+        if source == os.path.join(os.sep, "etc", "daos"):
+            # Use a temporary sub-directory in a directory where the user has permissions
+            tmp_copy_dir = os.path.join(
+                os.environ.get("DAOS_TEST_LOG_DIR", DEFAULT_DAOS_TEST_LOG_DIR), tmp_copy_dir)
+            sudo_command = "sudo -n "
+        else:
+            tmp_copy_dir = os.path.join(source, tmp_copy_dir)
+            sudo_command = ""
+
+        # Create a temporary remote directory
+        command = f"mkdir -p {tmp_copy_dir}"
+        if not run_remote(logger, hosts, command).passed:
+            message = f"Error creating temporary remote copy directory {tmp_copy_dir}"
+            self._fail_test(self.result.tests[-1], "Process", message)
+            return 16
+
+        # Move all the source files matching the pattern into the temporary remote directory
+        other = f"-print0 | xargs -0 -r0 -I '{{}}' {sudo_command}mv '{{}}' {tmp_copy_dir}/"
+        if not run_remote(logger, hosts, find_command(source, pattern, depth, other)).passed:
+            message = f"Error moving files to temporary remote copy directory {tmp_copy_dir}"
+            self._fail_test(self.result.tests[-1], "Process", message)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'fail_reason' in staticmethod call